core/src/main/scala/kafka/common/OffsetMetadataAndError.scala

```diff
@@ -44,7 +44,7 @@ case class OffsetAndMetadata(offsetMetadata: OffsetMetadata,
 }
 
 object OffsetAndMetadata {
-  def apply(offset: Long, metadata: String, commitTimestamp: Long, expireTimestamp: Long) = new OffsetAndMetadata(OffsetMetadata(offset, metadata), commitTimestamp, expireTimestamp)
+  def apply(offset: Long, metadata: String, commitTimestamp: Long, expireTimestamp: Long) = new OffsetAndMetadata(OffsetMetadata(offset, metadata), commitTimestamp, Some(expireTimestamp))
```

Comment:
Hmm.. Should we be checking for the default timestamp here? I had expected that we would use `None` if the default is used.

Reply:
Wouldn't the default be when the `expireTimestamp` argument is not provided? which will be taken care of by the following methods?

Reply:
I had assumed so initially, but then I saw the `contains(DEFAULT_TIMESTAMP)` check below.